python-sdk/tests_integration/databases/test_all_databases.py

```diff
@@ -148,6 +182,29 @@ def test_fetch_all_rows(mock_run_sql, database_table_fixture, row_count):
         assert f"LIMIT {row_count}" not in select_statements[0]
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {"database": Database.MSSQL},
+    ],
+    indirect=True,
+    ids=["mssql"],
+)
+@pytest.mark.parametrize("row_count", [0, 100])
+@mock.patch.object(BaseDatabase, "run_sql")
+def test_fetch_all_rows_mssql(mock_run_sql, database_table_fixture, row_count):
+    db, table = database_table_fixture
+    db.run_sql = mock_run_sql
+    db.fetch_all_rows(table, row_count)
+    select_statements = [m.args[0] for m in mock_run_sql.mock_calls if m.args and "SELECT" in m.args[0]]
+    assert len(select_statements) == 1
+    if row_count > -1:
+        assert f"TOP {row_count}" in select_statements[0]
+    else:
+        assert f"TOP {row_count}" not in select_statements[0]
```

Comment:
According to `@pytest.mark.parametrize("row_count", [0, 100])` the row count will always be above -1. Why do we need the if/else?

Reply:
fixed it